src/api_v1.py

```diff
@@ -36,18 +45,31 @@ def get(self):
         return {"status": "ok"}
 
 
-# class RunTrainingJob(Resource):
-#     def post(self):
-#         ecosystem = request.form.get('ecosystem')
-#         model = request.form.get('model')
-#         run_emr.connect()
-#         resp = run_emr.run_training_job(model, ecosystem)
-#         return resp, 200
+class RunTrainingJob(Resource):
+    """API for retraining purpose."""
+
+    def post(self):
+        """POST call for initiating retraining of models."""
+        required_fields = ["data_version", "bucket_name", "github_repo", "ecosystem"]
+        input_data = request.get_json()
+        missing_fileds = check_field_exists(input_data, required_fields)
+        print(missing_fileds)
+        if missing_fileds:
+            raise HTTPError(400, "Some field is missing in input data")
+        if not input_data:
+            raise HTTPError(400, "Expected JSON request")
+        input_data['environment'] = config.DEPLOYMENT_PREFIX
+        ecosystem = input_data.get('ecosystem')
+        emr_instance = emr_instances.get(ecosystem)
+        if emr_instance:
+            emr_instance = emr_instance()
+            status = emr_instance.run_job(input_data)
+        return status
```

Comment:
handle the `else` branch, what if emr_instance is `None` and you are returning `status`  that variable not defined. It will return `NameError`